frontend/src/pages/create_campaign/Roles/index.js

```diff
@@ -0,0 +1,297 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { v4 as uuidv4 } from "uuid";
+import {
+  ListItemText,
+  ListItemIcon,
+  IconButton,
+  Divider,
+  TextField,
+  Menu,
+  MenuItem,
+} from "@mui/material";
+import ClearIcon from "@mui/icons-material/Clear";
+import EditIcon from "@mui/icons-material/Edit";
+import AddIcon from "@mui/icons-material/Add";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import {
+  ContainerDiv,
+  QuestionsDiv,
+  RolesDisplay,
+  QuestionsDisplay,
+  RolesDiv,
+  RoleListItem,
+  RoleListItemButton,
+  QuestionsHeader,
+  SectionTitle,
+  RolesListContainer,
+  RoleQuantity,
+  CreateRoleFormControl,
+  CreateRoleFormGroup,
+  CreateRoleQuantity,
+  CreateRoleName,
+  AddQuestionButton,
+  SelectFromExistingButton,
+  QuestionTitle,
+  QuestionContent,
+} from "./rolesTab.styled";
+
+const RolesTab = (props) => {
+  const {
+    value,
+    index,
+    roles,
+    setRoles,
+    questions,
+    setQuestions,
+    roleSelected,
+    setRoleSelected,
+  } = props;
+  const [newRoleName, setNewRoleName] = useState("");
+  const [newRoleQty, setNewRoleQty] = useState(1);
+  const [anchorEl, setAnchorEl] = useState(null);
+  const open = Boolean(anchorEl);
```

Comment:
i'd rather have just one variable representing one state - right now we have two which might be confusing later on? i think its okay to have `Boolean(anchorEl)` in your code if you really need